img_processing/src/detector.cpp

```
@@ -0,0 +1,255 @@
+{
+    vector<bool> isValid(allChanges.size(), true);
+    for (int i = 0; i < allChanges.size(); i++) {
+        if (isValid[i]) {
+            for (int j = 0; j < allChanges.size(); j++) {
+                if (isValid[j]) {
+                    if (i != j && (allChanges[i] & allChanges[j]).area() > 0) {
+                        if (i > j) {
+                            allChanges[i] |= allChanges[j];
+                            isValid[j] = false;
+                        }
+                        else {
+                            allChanges[j] |= allChanges[i];
+                            isValid[i] = false;
+                        }
+                    }
+                }
+            }
+        }
+    }
+    for (int i = 0; i < allChanges.size(); i++) {
+        if (!isValid[i]) {
+            isValid.erase(isValid.begin() + i);
+            allChanges.erase(allChanges.begin() + i);
+            i--;
+        }
+    }
+    for (Rect r : allChanges) {
+        rectangle(*currentFrame, r, Scalar(0, 0, 255), 2);
+    }
+    imshow("win", *currentFrame);
+    waitKey();
+    return allChanges;
+}
+
+vector<ObjectInformation> Detector::getOutput() const
+{
+    return output;
+}
+
+// A function that changes the image to fit the model
+Mat Detector::formatYolov5(const shared_ptr<Mat> &frame)
+{
+    int col = frame->cols;
+    int row = frame->rows;
+    int maximum = MAX(col, row);
+    Mat result = Mat::zeros(maximum, maximum, CV_8UC3);
+    frame->copyTo(result(Rect(0, 0, col, row)));
+    return result;
+}
+
+void Detector::init(bool isCuda)
+{
+    loadNet(isCuda);
+}
+
+// Loading yolov5s onnx model
+void Detector::loadNet(bool isCuda)
+{
+    auto result = readNet("../yolov5s.onnx");
+    if (result.empty()) {
+        Manager::imgLogger.logMessage(logger::LogLevel::ERROR,
+                                      "failed to load yolov5 model");
+    }
+
+    if (result.empty()) {
```

you have the same check twice,
Also what is the right behavior in case of an error?
Can we make continue? or we should return an error?